### ISSUE TITLE
Archive reading

### ISSUE THREAD

**Issue by GenoXu** [mienaiyami/yomikiru#90]
### Description of the new feature / enhancement

If you could make it read zipped files directly (like in other manga readers). It's very inconvenient to wait for unzipping every time.

**Comment by mienaiyami** [comment]
Read zipped files without unzipping? 
Not sure how but I will look into it after #78.

**Comment by GenoXu** [comment]
Something like this:

https://user-images.githubusercontent.com/31614025/230155797-b740670a-cb33-4173-bb8e-26f83d4e7b5b.mp4

**Comment by mienaiyami** [comment]
Maybe it is unzipping it as well, try comparing its speed to apps unzipping time

**Comment by GenoXu** [comment]
I have a slow machine, but the difference is so remarkable!
<br>

https://user-images.githubusercontent.com/31614025/230159094-84cd9492-90d8-40aa-a9e1-11247b4c218f.mp4

<br>
Maybe the other readers display each image just when it's unzipped, not after finishing the whole archive(?)

**Comment by GenoXu** [comment]
Look how it keeps loading images after starting
<br>

https://user-images.githubusercontent.com/31614025/230161435-0851f13c-c889-4d6c-b28f-ebfe9f6fb357.mp4

**Comment by GenoXu** [comment]
<br>


https://user-images.githubusercontent.com/31614025/230163284-3ff2e720-376e-416d-b9b1-3b25a66c91a8.mp4

**Comment by mienaiyami** [comment]
Sorry for this type of image loading system, I will try to improve it for best experience and speed.

**Comment by mienaiyami** [comment]
Additionally, in case you dont know this app initially supported and was made for only normal folder (not archive) in mind so this is cause coz this type of system.

**Comment by mienaiyami** [comment]
Im planning to add option disable image loading screen.

**Comment by mienaiyami** [comment]
I meant with this option enabled, reader will not wait for all images to load and load images dynamically. And yes it might cause stutter but some are fine with that.

**Comment by GenoXu** [comment]
@mienaiyami
I want to confirm that my main problem is with the _unzipping_ process; my zip app took 2s to unzip a file but this reader took more than 40s for the same file (thus, I suggested the direct reading thing)!
I'm fine with loading all images at once IF the unzipping was faster.

https://user-images.githubusercontent.com/31614025/230394439-b56aa7e4-8f69-4afb-a464-f36581a87357.mp4


<br>
<br>

> Im planning to add option disable image loading screen.

If this is ever implemented, just make it disableable for people who are not fine with high RAM usage.

**Comment by mienaiyami** [comment]
Thanks, I will look into what's causing low unzipping, maybe my app's unzipper is the issue.

**Comment by mienaiyami** [comment]
I think there is not anything else I can do in this matter. I tried to make unzipper faster. You can also try changing temp folder in settings added in v2.15.0, might help. Also, I still can't reproduce it.

**Comment by GenoXu** [comment]
Sorry for the late comment.
Now (v2.15.0) it takes only about 4s to be unzipped. Thank you.

**Comment by GenoXu** [comment]
@mienaiyami A trivial matter (not worth opening a new issue): the warning before closing the app pops up twicely

https://github.com/mienaiyami/yomikiru/assets/31614025/3ff3c452-c13d-459c-81fd-e7753ed01ff9

**Comment by mienaiyami** [comment]
Wasn't a trivial matter at all. Had to redo many things but thanks to that few other related issues got fixed as well.
Thanks for reporting.

**Comment by GenoXu** [comment]
Now, it doesn't pop up at all :)

https://github.com/mienaiyami/yomikiru/assets/31614025/00cf89c5-5445-430d-94f4-a6e3d5e247e3

**Comment by mienaiyami** [comment]
Thats very weird, i tested it before releasing update. And it might be 3rd time today that it local build run well but doesnt after release.
Well get ready for 3rd update in same day. 
Sorry about that.
